crates/services/block_aggregator_api/src/blocks/importer_and_db_source/sync_service.rs

```diff
@@ -98,6 +99,12 @@ where
         }
         Ok(txs)
     }
+
+    // For now just have arbitrary 10 ms sleep to avoid busy looping.
+    // This could be more complicated with increasing backoff times, etc.
+    async fn go_to_sleep_before_continuing(&self) {
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
 }
 
 impl<Serializer, DB, E> RunnableTask for SyncTask<Serializer, DB>
```

Comment:
[nit] https://github.com/FuelLabs/fuel-rust-indexer/blob/main/crates/receipts_manager/service.rs#L584

We can replace the whole service with stream of joined events(old + new)